src/test/java/org/prebid/server/bidder/rubicon/RubiconBidderTest.java

```diff
@@ -192,6 +198,35 @@ public void makeHttpRequestsShouldTolerateInvalidSizes() {
                 .containsOnly(15);
     }
 
+    @Test
+    public void makeHttpRequestsShouldOverrideBannerFormatWithRubiconSizes() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(singletonList(Imp.builder()
+                        .banner(Banner.builder()
+                                .format(asList(
+                                        Format.builder().w(300).h(250).build(),
+                                        Format.builder().w(300).h(600).build()))
+                                .build())
+                        .ext(mapper.valueToTree(ExtPrebid.of(null, ExtImpRubicon.builder()
+                                .sizes(singletonList(15)).build())))
+                        .build()))
+                .build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = rubiconBidder.makeHttpRequests(bidRequest);
+
+        // then
+        final List<Format> expectedFormat = singletonList(Format.builder().w(300).h(250).build());
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1).doesNotContainNull()
+                .extracting(httpRequest -> mapper.readValue(httpRequest.getBody(), BidRequest.class))
+                .flatExtracting(BidRequest::getImp).doesNotContainNull()
+                .extracting(Imp::getBanner).doesNotContainNull()
+                .extracting(Banner::getFormat).hasSize(1)
+                .contains(expectedFormat, Index.atIndex(0));
```

Comment:
We can use `containsOnly(...)` method because expected only 1 element in `banner.format` collection.